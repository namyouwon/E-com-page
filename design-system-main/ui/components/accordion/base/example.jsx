// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import uniqueId from 'lodash.uniqueid';
import Accordion, { AccordionSection } from '../';
import { ActionOverflow } from '../../../components/menus/action-overflow/example';
import Card from '../../cards/';

export const AccordionExample = () => (
  <Accordion>
    <AccordionSection
      summary="Accordion summary"
      isOpen
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - A
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - B
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - C
    </AccordionSection>
  </Accordion>
);

export default <AccordionExample />;

export let states = [
  {
    id: 'section-one-open',
    label: 'Section one open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-two-open',
    label: 'Section two open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-three-open',
    label: 'Section three open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'multiple-sections-open',
    label: 'Multiple sections open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];

export let examples = [
  {
    id: 'long-heading-truncated',
    label: 'Long Section Heading with Truncation for larger screens',
    demoStyles: 'max-width: 500px;',
    storybookStyles: true,
    element: (
      <Card>
        <Accordion>
          <AccordionSection
            summary="Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </Card>
    )
  },
  {
    id: 'nested-accordions',
    label: 'Nested Accordions',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          <Accordion>
            <AccordionSection
              summary="Nested accordion summary"
              isOpen
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - A
            </AccordionSection>
            <AccordionSection
              summary="Nested accordion summary"
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - B
            </AccordionSection>
          </Accordion>
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'styled',
    label: 'Wrapped in Card',
    element: (
      <div className="slds-card">
        <Accordion>
          <AccordionSection
            summary="Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </div>
    )
  },
  {
    id: 'has-action-overflow',
    label: 'Has Action Overflow Menu',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Background Color',
    demoStyles: '--slds-c-accordion-summary-color-background: lightblue',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Text Color',
    demoStyles:
      '--slds-c-accordion-heading-text-color: green; --slds-c-accordion-heading-text-color-hover: red',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];
