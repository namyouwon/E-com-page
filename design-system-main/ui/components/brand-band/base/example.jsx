// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';

import BrandBand from '../';

export default <BrandBand size="medium" />;

export let examples = [
  {
    id: 'brand-band-small',
    label: 'Small',
    element: <BrandBand size="small" />
  },
  {
    id: 'brand-band-large',
    label: 'Large',
    element: <BrandBand size="large" />
  },
  {
    id: 'brand-band-no-image',
    label: 'No Image',
    element: <BrandBand size="none" />
  },
  {
    id: 'brand-band-large-group',
    label: 'Large Group',
    element: <BrandBand size="large" image="group" />
  },
  {
    id: 'brand-band-large-user',
    label: 'Large User',
    element: <BrandBand size="large" image="user" />
  }
];
