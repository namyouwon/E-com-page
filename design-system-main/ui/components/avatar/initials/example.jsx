// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Avatar>
        <AvatarInitials
          iconClassName="slds-icon-standard-account"
          title="company name"
          initials="Ac"
        />
      </Avatar>
    )
  }
];

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="#" onClick={e => e.preventDefault()}>
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
