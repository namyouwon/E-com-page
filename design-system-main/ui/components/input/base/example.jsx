// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import PropTypes from 'prop-types';
import Input from '../';
import SvgIcon from '../../../shared/svg-icon';
import ButtonIcon from '../../button-icons/';
import { Spinner } from '../../spinners/base/example';
import {
  FormElement,
  SimpleFormElementWrapper,
  FormElementControl
} from '../../form-element';
import uniqueId from 'lodash.uniqueid';

const inputLabel = 'Input Label';
const placeholderText = 'Placeholder textâ€¦';

let Required = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement labelContent={inputLabel} inputId={inputId} isRequired>
      <Input id={inputId} placeholder={placeholderText} required />
    </FormElement>
  );
};

let ErrorState = () => {
  const inputId = uniqueId('text-input-id-');
  const errorId = uniqueId('error-message-id-');

  return (
    <FormElement
      hasError
      labelContent={inputLabel}
      inputId={inputId}
      errorId={errorId}
      isRequired
      inlineMessage="Enter a value."
    >
      <Input
        id={inputId}
        placeholder={placeholderText}
        required
        aria-describedby={errorId}
      />
    </FormElement>
  );
};

let ErrorIcon = () => {
  const inputId = uniqueId('text-input-id-');
  const errorId = uniqueId('error-message-id-');

  return (
    <FormElement
      hasError
      labelContent={inputLabel}
      inputId={inputId}
      hasLeftIcon
      errorId={errorId}
      isRequired
      inlineMessage="Enter a value."
    >
      <SvgIcon className="slds-input__icon" sprite="utility" symbol="error" />
      <Input
        id={inputId}
        required
        placeholder={placeholderText}
        aria-describedby={errorId}
      />
    </FormElement>
  );
};

let Disabled = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement labelContent={inputLabel} inputId={inputId}>
      <Input id={inputId} placeholder={placeholderText} disabled />
    </FormElement>
  );
};

let Readonly = props => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement labelContent={inputLabel} inputId={inputId}>
      <Input
        id={inputId}
        readOnly
        hasBorders={props.hasBorders}
        defaultValue="Read Only"
        placeholder=""
      />
    </FormElement>
  );
};

let LeftIcon = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement labelContent={inputLabel} inputId={inputId} hasLeftIcon>
      <SvgIcon
        className="slds-icon slds-input__icon slds-input__icon_left slds-icon-text-default"
        sprite="utility"
        symbol="search"
      />
      <Input id={inputId} placeholder={placeholderText} />
    </FormElement>
  );
};

let RightIcon = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement labelContent={inputLabel} inputId={inputId} hasRightIcon>
      <SvgIcon
        className="slds-icon slds-input__icon slds-input__icon_right slds-icon-text-default"
        sprite="utility"
        symbol="search"
      />
      <Input id={inputId} placeholder={placeholderText} />
    </FormElement>
  );
};

let DoubleIcon = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement
      labelContent={inputLabel}
      inputId={inputId}
      hasLeftIcon
      hasRightIcon
    >
      <SvgIcon
        className="slds-icon slds-input__icon slds-input__icon_left slds-icon-text-default"
        sprite="utility"
        symbol="search"
      />
      <Input id={inputId} placeholder={placeholderText} />
      <ButtonIcon
        symbol="clear"
        className="slds-input__icon slds-input__icon_right"
        iconClassName="slds-icon-text-light"
        assistiveText="Clear"
        title="Clear"
      />
    </FormElement>
  );
};

let DoubleIconSpinner = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement
      labelContent={inputLabel}
      inputId={inputId}
      hasLeftIcon
      hasRightIcon
      hasRightIconGroup
    >
      <SvgIcon
        className="slds-icon slds-input__icon slds-input__icon_left slds-icon-text-default"
        sprite="utility"
        symbol="search"
      />
      <Input id={inputId} placeholder={placeholderText} />
      <div className="slds-input__icon-group slds-input__icon-group_right">
        <Spinner className="slds-spinner_brand slds-spinner_x-small slds-input__spinner" />
        <ButtonIcon
          symbol="clear"
          className="slds-input__icon slds-input__icon_right"
          iconClassName="slds-icon-text-light"
          assistiveText="Clear"
          title="Clear"
        />
      </div>
    </FormElement>
  );
};

let InputWithType = props => {
  const inputId = uniqueId(`${props.type}-input-id-`);

  return (
    <FormElement labelContent={props.label} inputId={inputId}>
      <Input id={inputId} type={props.type} placeholder={props.placeholder} />
    </FormElement>
  );
};

InputWithType.propTypes = {
  type: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string
};

InputWithType.defaultProps = {
  label: 'Input Label'
};

let FixedText = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement
      labelContent={inputLabel}
      inputId={inputId}
      formControlClassName="slds-input-has-fixed-addon"
      labelId="fixed-text-label"
    >
      <span className="slds-form-element__addon" id="fixed-text-addon-pre">
        $
      </span>
      <Input
        id={inputId}
        placeholder={placeholderText}
        aria-labelledby="fixed-text-label fixed-text-addon-pre fixed-text-addon-post"
      />
      <span className="slds-form-element__addon" id="fixed-text-addon-post">
        euro
      </span>
    </FormElement>
  );
};

let InlineHelp = () => {
  const inputId = uniqueId('text-input-id-');
  const errorId = uniqueId('error-message-id-');

  return (
    <FormElement
      labelId="inline-text-label"
      labelContent={inputLabel}
      inputId={inputId}
      errorId={errorId}
      inlineMessage="ex: (415)111-2222"
    >
      <Input
        id={inputId}
        placeholder={placeholderText}
        aria-describedby={errorId}
      />
    </FormElement>
  );
};

const FieldLevelHelp = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement
      labelContent={inputLabel}
      inputId={inputId}
      hasTooltip
      showTooltip
    >
      <Input id={inputId} placeholder={placeholderText} />
    </FormElement>
  );
};

const Bare = () => {
  const inputId = uniqueId('text-input-id-');

  return (
    <FormElement labelContent={inputLabel} inputId={inputId}>
      <Input id={inputId} isBare placeholder={placeholderText} />
    </FormElement>
  );
};

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <FormElement labelContent={inputLabel} inputId="text-input-id-0">
        <Input id="text-input-id-0" placeholder={placeholderText} />
      </FormElement>
    )
  }
];

export let states = [
  {
    id: 'input-required',
    label: 'Required',
    element: <Required />
  },
  {
    id: 'input-disabled',
    label: 'Disabled',
    element: <Disabled />
  },
  {
    id: 'input-error',
    label: 'Error',
    element: <ErrorState />
  },
  {
    id: 'input-error-icon',
    label: 'Error with icon',
    element: <ErrorIcon />
  },
  {
    id: 'read-only',
    label: 'Readonly',
    element: <Readonly />
  },
  {
    id: 'read-only-borders',
    label: 'Readonly with borders',
    element: <Readonly hasBorders />
  },
  {
    id: 'static',
    label: 'Static',
    element: (
      <SimpleFormElementWrapper>
        <span className="slds-form-element__label">Input Label</span>
        <FormElementControl>
          <span className="slds-form-element__static">Read Only</span>
        </FormElementControl>
      </SimpleFormElementWrapper>
    )
  }
];

export let examples = [
  {
    id: 'left-icon',
    label: 'Left Icon',
    element: <LeftIcon />
  },
  {
    id: 'right-icon',
    label: 'Right Icon',
    element: <RightIcon />
  },
  {
    id: 'double-icon',
    label: 'Left Icon & Clear Button',
    element: <DoubleIcon />
  },
  {
    id: 'double-icon-spinner',
    label: 'Clear Button with Spinner',
    element: <DoubleIconSpinner />
  },
  {
    id: 'email-input',
    label: 'Email Input',
    element: <InputWithType type="email" label="Email" placeholder="Email" />
  },
  {
    id: 'date-input',
    label: 'Date Input',
    element: <InputWithType type="date" />
  },
  {
    id: 'search-input',
    label: 'Search Input',
    element: <InputWithType type="search" />
  },
  {
    id: 'url-input',
    label: 'URL Input',
    element: <InputWithType type="url" />
  },
  {
    id: 'tel-input',
    label: 'Telephone Input',
    element: <InputWithType type="tel" />
  },
  {
    id: 'fixed-text',
    label: 'Fixed text',
    element: <FixedText />
  },
  {
    id: 'inline-help',
    label: 'Inline Help',
    element: <InlineHelp />
  },
  {
    id: 'field-level-help',
    label: 'Field level help',
    demoStyles: `
      padding-top: 3rem;
      position: relative;
    `,
    storybookStyles: `
      padding-top: 4rem;
      position: relative;
    `,
    element: <FieldLevelHelp />
  },
  {
    id: 'bare',
    label: 'Bare',
    element: <Bare />
  }
];
