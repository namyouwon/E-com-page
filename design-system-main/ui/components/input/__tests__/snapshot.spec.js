/* eslint-env jest */
import React from 'react';
import { getDisplayElementById } from '../../../shared/helpers';
import * as Input from '../base/example';
import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

describe('Input', () => {
  it('renders a default input', () => matchesMarkup(Input.default));

  it('renders an input that is required', () =>
    matchesMarkup(getDisplayElementById(Input.states, 'input-required')));

  it('renders an input that is disabled', () =>
    matchesMarkup(getDisplayElementById(Input.states, 'input-disabled')));

  it('renders an input that has an error', () =>
    matchesMarkup(getDisplayElementById(Input.states, 'input-error')));

  it('renders an input that has an error with icon', () =>
    matchesMarkup(getDisplayElementById(Input.states, 'input-error-icon')));

  it('renders an input that is read only', () =>
    matchesMarkup(getDisplayElementById(Input.states, 'read-only')));

  it('renders an input with left icon', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'left-icon')));

  it('renders an input with right icon', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'right-icon')));

  it('renders an input with left icon and clear button', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'double-icon')));

  it('renders an input with clear button and spinner', () =>
    matchesMarkup(
      getDisplayElementById(Input.examples, 'double-icon-spinner')
    ));

  it('renders an input that is the email type', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'email-input')));

  it('renders an input that is the date type', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'date-input')));

  it('renders an input that is the search type', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'search-input')));

  it('renders an input that is the url type', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'url-input')));

  it('renders an input that is the tel type', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'tel-input')));

  it('renders an input with fixed text', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'fixed-text')));

  it('renders an input with inline help', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'inline-help')));

  it('renders an input with field level help', () =>
    matchesMarkup(getDisplayElementById(Input.examples, 'field-level-help')));
});
