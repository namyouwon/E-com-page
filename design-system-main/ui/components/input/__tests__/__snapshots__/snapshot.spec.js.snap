// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders a default input 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-0"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="text-input-id-0"
      placeholder="Placeholder text…"
      type="text"
    />
  </div>
</div>
`;

exports[`Input renders an input that has an error 1`] = `
<div
  className="slds-form-element slds-has-error"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-3"
  >
    <abbr
      className="slds-required"
      title="required"
    >
      *
       
    </abbr>
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      aria-describedby="error-message-id-4"
      aria-invalid="true"
      className="slds-input"
      id="text-input-id-3"
      placeholder="Placeholder text…"
      required={true}
      type="text"
    />
  </div>
  <div
    className="slds-form-element__help"
    id="error-message-id-4"
  >
    Enter a value.
  </div>
</div>
`;

exports[`Input renders an input that has an error with icon 1`] = `
<div
  className="slds-form-element slds-has-error"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-5"
  >
    <abbr
      className="slds-required"
      title="required"
    >
      *
       
    </abbr>
    Input Label
  </label>
  <div
    className="slds-form-element__control slds-input-has-icon slds-input-has-icon_left"
  >
    <svg
      aria-hidden={true}
      className="slds-input__icon"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#error"
      />
    </svg>
    <input
      aria-describedby="error-message-id-6"
      aria-invalid="true"
      className="slds-input"
      id="text-input-id-5"
      placeholder="Placeholder text…"
      required={true}
      type="text"
    />
  </div>
  <div
    className="slds-form-element__help"
    id="error-message-id-6"
  >
    Enter a value.
  </div>
</div>
`;

exports[`Input renders an input that is disabled 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-2"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      disabled={true}
      id="text-input-id-2"
      placeholder="Placeholder text…"
      type="text"
    />
  </div>
</div>
`;

exports[`Input renders an input that is read only 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-7"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      defaultValue="Read Only"
      id="text-input-id-7"
      placeholder=""
      readOnly={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Input renders an input that is required 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-1"
  >
    <abbr
      className="slds-required"
      title="required"
    >
      *
       
    </abbr>
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="text-input-id-1"
      placeholder="Placeholder text…"
      required={true}
      type="text"
    />
  </div>
</div>
`;

exports[`Input renders an input that is the date type 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="date-input-id-13"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="date-input-id-13"
      type="date"
    />
  </div>
</div>
`;

exports[`Input renders an input that is the email type 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="email-input-id-12"
  >
    Email
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="email-input-id-12"
      placeholder="Email"
      type="email"
    />
  </div>
</div>
`;

exports[`Input renders an input that is the search type 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="search-input-id-14"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="search-input-id-14"
      type="search"
    />
  </div>
</div>
`;

exports[`Input renders an input that is the tel type 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="tel-input-id-16"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="tel-input-id-16"
      type="tel"
    />
  </div>
</div>
`;

exports[`Input renders an input that is the url type 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="url-input-id-15"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="url-input-id-15"
      type="url"
    />
  </div>
</div>
`;

exports[`Input renders an input with clear button and spinner 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-11"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control slds-input-has-icon slds-input-has-icon_left-right slds-input-has-icon_group-right"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-input__icon slds-input__icon_left slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#search"
      />
    </svg>
    <input
      className="slds-input"
      id="text-input-id-11"
      placeholder="Placeholder text…"
      type="text"
    />
    <div
      className="slds-input__icon-group slds-input__icon-group_right"
    >
      <div
        className="slds-spinner slds-spinner_brand slds-spinner_x-small slds-input__spinner"
        role="status"
      >
        <span
          className="slds-assistive-text"
        >
          Loading
        </span>
        <div
          className="slds-spinner__dot-a"
        />
        <div
          className="slds-spinner__dot-b"
        />
      </div>
      <button
        className="slds-button slds-button_icon slds-input__icon slds-input__icon_right"
        onClick={[Function]}
        title="Clear"
      >
        <svg
          aria-hidden={true}
          className="slds-button__icon slds-icon-text-light"
        >
          <use
            xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#clear"
          />
        </svg>
        <span
          className="slds-assistive-text"
        >
          Clear
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Input renders an input with field level help 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-20"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__icon"
  >
    <button
      aria-describedby="help"
      className="slds-button slds-button_icon"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#info"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Help
      </span>
    </button>
    <div
      className="slds-popover slds-popover_tooltip slds-nubbin_bottom-left"
      id="help"
      role="tooltip"
      style={
        Object {
          "left": "-15px",
          "position": "absolute",
          "top": "-45px",
          "width": "170px",
        }
      }
    >
      <div
        className="slds-popover__body"
      >
        Some helpful information.
      </div>
    </div>
  </div>
  <div
    className="slds-form-element__control"
  >
    <input
      className="slds-input"
      id="text-input-id-20"
      placeholder="Placeholder text…"
      type="text"
    />
  </div>
</div>
`;

exports[`Input renders an input with fixed text 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-17"
    id="fixed-text-label"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control slds-input-has-fixed-addon"
  >
    <span
      className="slds-form-element__addon"
      id="fixed-text-addon-pre"
    >
      $
    </span>
    <input
      aria-labelledby="fixed-text-label fixed-text-addon-pre fixed-text-addon-post"
      className="slds-input"
      id="text-input-id-17"
      placeholder="Placeholder text…"
      type="text"
    />
    <span
      className="slds-form-element__addon"
      id="fixed-text-addon-post"
    >
      euro
    </span>
  </div>
</div>
`;

exports[`Input renders an input with inline help 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-18"
    id="inline-text-label"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control"
  >
    <input
      aria-describedby="error-message-id-19"
      aria-invalid="true"
      className="slds-input"
      id="text-input-id-18"
      placeholder="Placeholder text…"
      type="text"
    />
  </div>
  <div
    className="slds-form-element__help"
    id="error-message-id-19"
  >
    ex: (415)111-2222
  </div>
</div>
`;

exports[`Input renders an input with left icon 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-8"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control slds-input-has-icon slds-input-has-icon_left"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-input__icon slds-input__icon_left slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#search"
      />
    </svg>
    <input
      className="slds-input"
      id="text-input-id-8"
      placeholder="Placeholder text…"
      type="text"
    />
  </div>
</div>
`;

exports[`Input renders an input with left icon and clear button 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-10"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control slds-input-has-icon slds-input-has-icon_left-right"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-input__icon slds-input__icon_left slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#search"
      />
    </svg>
    <input
      className="slds-input"
      id="text-input-id-10"
      placeholder="Placeholder text…"
      type="text"
    />
    <button
      className="slds-button slds-button_icon slds-input__icon slds-input__icon_right"
      onClick={[Function]}
      title="Clear"
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon slds-icon-text-light"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#clear"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Clear
      </span>
    </button>
  </div>
</div>
`;

exports[`Input renders an input with right icon 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-form-element__label"
    htmlFor="text-input-id-9"
  >
    Input Label
  </label>
  <div
    className="slds-form-element__control slds-input-has-icon slds-input-has-icon_right"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-input__icon slds-input__icon_right slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#search"
      />
    </svg>
    <input
      className="slds-input"
      id="text-input-id-9"
      placeholder="Placeholder text…"
      type="text"
    />
  </div>
</div>
`;
